sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -247,6 +369,16 @@ def incr_mean_variance_axis0(X, last_mean, last_var, unsigned long last_n):
                                      X.format, last_mean, last_var, last_n)
 
 
+def incr_mean_variance_axis0_nan(X, last_mean, last_var, unsigned long last_n):
+    """This is identical to incr_mean_variance_axis0_nan
+    but can process NaN values
+    """
+    if X.dtype != np.float32:
```

Comment:
@jnothman do we convert here because we are scared about overflow or this is an unnecessary thing?